packages/public-api/src/website/metadata/favicon.ts

```diff
@@ -12,9 +15,33 @@ export async function saveScanWebPageFavicon(
     return;
   }
 
-  const faviconRequest = await fetch(pageMetadata.favicon);
+  abortController = new AbortController();
+
+  const faviconRequest = await Promise.race([
+    fetch(pageMetadata.favicon, {
+      headers: { Accept: VALID_MIME_TYPES.join(', ') },
+      // TODO: remove ignores when typescript versions are synced
+      /* eslint-disable-next-line @typescript-eslint/prefer-ts-expect-error */
+      // @ts-ignore note: node-fetch typings are incompatible
+      signal: abortController.signal,
+    }),
+    new Promise<null>((resolve) => {
+      setTimeout(() => {
+        abortController.abort();
+        resolve(null);
+      }, FAVICON_TIMEOUT);
+    }),
+  ]);
+
+  if (!faviconRequest) {
+    return;
+  }
 
   const faviconContentType = faviconRequest.headers.get('content-type') ?? undefined;
+  if (faviconContentType && !VALID_MIME_TYPES.includes(faviconContentType)) {
```

Comment:
Might be better to be strict here and return if the `content-type` header is missing.